jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1117,25 +1137,31 @@ public void write(boolean last, ByteBuffer content, Callback callback)
         {
             long length = BufferUtil.length(content);
 
-            HttpChannelState httpChannel;
+            HttpChannelState httpChannelState;
             HttpStream stream;
             Throwable writeFailure;
             MetaData.Response responseMetaData = null;
             try (AutoLock ignored = _request._lock.lock())
             {
-                httpChannel = _request.lockedGetHttpChannelState();
+                httpChannelState = _request.lockedGetHttpChannelState();
                 long totalWritten = _contentBytesWritten + length;
                 writeFailure = _writeFailure;
 
                 if (writeFailure == null)
                 {
                     if (_writeCallback != null)
                     {
+                        if (_writeCallback instanceof InterimCallback interimCallback)
+                        {
+                            // Do this write after the interim callback.
+                            interimCallback.thenRun(() -> write(last, content, callback));
```

Comment:
Again, `thenRun()` only runs if CF succeeds. If fails don't we have to at least fail the write callback?